apps/snitch_core/priv/repo/migrations/20181127092015_update_product_state_type.exs

```diff
@@ -0,0 +1,79 @@
+defmodule Snitch.Repo.Migrations.UpdateProductStateType do
+  use Ecto.Migration
+  import Ecto.Query
+
+  alias Snitch.Data.Schema
+  alias Snitch.Repo
+
+  def up do
+    state_map =
+      StateEnum.__enum_map__()
+      |> Enum.into(%{}, fn x -> {elem(x, 0) |> Atom.to_string(), elem(x, 1)} end)
+
+    id_state_map =
+      Repo.all(from(p in "snitch_products", select: %{id: p.id, state: p.state}))
+      |> Enum.into([], fn x -> %{id: x.id, state: Map.get(state_map, x.state, 0)} end)
```

Comment:
`Repo.all` return Enumerable, so we can directy use `Enum.map` instead of `Enum.into`

Reply:
done